exchanges/persisted-fetch/src/persistedFetchExchange.ts

```diff
@@ -71,22 +71,32 @@ export const persistedFetchExchange = (
           // Hash the given GraphQL query
           fromPromise(hashFn(query, operation.query)),
           mergeMap(sha256Hash => {
-            // Attach SHA256 hash and remove query from body
-            body.query = undefined;
-            body.extensions = {
-              persistedQuery: {
-                version: 1,
-                sha256Hash,
-              },
-            };
-
-            return makePersistedFetchSource(
-              operation,
-              body,
-              dispatchDebug,
-              !!(options as PersistedFetchExchangeOptions)
-                .preferGetForPersistedQueries
-            );
+            // if the hashing operation was successful, add the persisted query extension
+            if (sha256Hash) {
+              // Attach SHA256 hash and remove query from body
+              body.query = undefined;
+              body.extensions = {
+                persistedQuery: {
+                  version: 1,
+                  sha256Hash,
+                },
+              };
+
+              return makePersistedFetchSource(
+                operation,
+                body,
+                dispatchDebug,
+                !!(options as PersistedFetchExchangeOptions)
+                  .preferGetForPersistedQueries
+              );
+            } else {
+              return makePersistedFetchSource(
+                operation,
+                body,
+                dispatchDebug,
+                false
```

Comment:
We could move up the if-block and let this condition be `!!((options as PersistedFetchExchangeOptions)
                  .preferGetForPersistedQueries && sha256Hash)` 🤔  may not be worth it though

Reply:
Good idea, that's definitely cleaner!